templates/CRM/Financial/Form/Search.tpl

```diff
@@ -242,6 +242,8 @@ CRM.$(function($) {
         if (response.status == 'record-updated-success') {
           CRM.alert(listRecords(records), op == 'delete' ? {/literal}'{ts escape="js"}Deleted{/ts}' : '{ts escape="js"}Updated{/ts}'{literal}, 'success');
           batchSelector.fnDraw();
+	  //Redirect CRM-18169
+          window.location.href = CRM.url('civicrm/financial/financialbatches', 'reset=1&batchStatus=' + response.status_id);
```

Comment:
The request was to apply the filter to the batches again after re-opening/closing it so that it recently re-opened batch would be visible. I agree that the datatable shouldn't be refreshed if we are redirecting. I will change it.